### PR TITLE
chore: put collective ownership on most CODEOWNERS

### PR DESCRIPTION
**Pull request by thehowl**
This PR comes following a discussion in today's "review meeting", where we talked about our current approach to ownership.

One aspect which was talked about was that the current set-up for CODEOWNERS is very "noisy". Opening up a PR often entails tagging many people and teams for review. To try to overcome this, I removed most rules which tried to find "exact people" to own a piece of the code, in favour of collective ownership on the `tech-staff`, `security`, `devrels` and `devops` teams, depending on which part of the code is interested.

I wrote a few lines explaining the differences between assigning a directory to a team, a person, or a team + person, and how the CODEOWNERS file should be interpreted. I wrote them trying to match the practical implications of the CODEOWNERS file (= who gets picked in the requested reviewers) with the "social" / organizational implications.

### CODE REVIEW

**Comment by leohhhn** [comment]
Just to set expectations upfront: @gnolang/devrels are not codeowners of the docs since some time back - check out [this comment](https://github.com/gnolang/docs.gno.land/pull/45#issuecomment-2189394007).

However, I don't mind providing inputs.

File: .github/CODEOWNERS
Comment (leohhhn) [inline .github/CODEOWNERS:61]: can you add 
`/examples/gno.land/r/leon/` for me please? so I don't have to make a separate PR
```suggestion
/examples/gno.land/r/morgan/          @thehowl
/examples/gno.land/r/leon/                @leohhhn
```

**Review by moul — CHANGES_REQUESTED** [review]
Interesting! Let's discuss it during the upcoming retreat before we merge.

**Comment by sw360cab** [comment]
I would suggest adding `@gnolang/devops` to `.github/workflows`

File: .github/CODEOWNERS
Comment (Kouteki) [inline .github/CODEOWNERS:11]: Should we suggest to the individual code owner to unassign themselves if the change is small enough?

**Comment by thehowl** [comment]
This PR will be pending until @aeddi figures out a bot for github which will have a wider scope, and cover some of the things the CODEOWNERS file was currently attempting to do :)

**Comment by thehowl** [comment]
Looking to actually [remove CODEOWNERS](https://github.com/gnolang/gno/issues/1007#issuecomment-2427655633)

**Pull request closed without merging**
